extension/src/experiments/model/index.ts

```diff
@@ -829,4 +837,42 @@ export class ExperimentsModel extends ModelWithPersistence {
     }
     return commitsBySha
   }
+
+  private applyFiltersToFlattenedCommits() {
+    const commitsBySha: { [sha: string]: Commit[] } = {}
+    const filters = this.getFilters()
+
+    for (const commit of this.commits) {
+      const commitWithSelectedAndStarred = this.addDetails(commit)
+      const experiments = this.getExperimentsByCommit(
+        commitWithSelectedAndStarred
+      )
+
+      commitsBySha[commit.sha as string] = [
+        commitWithSelectedAndStarred,
+        ...(experiments || [])
+      ].filter(exp => !!filterExperiment(filters, exp))
+    }
+
+    return commitsBySha
+  }
+
+  private getFlattenedRowData(workspaceRow: Commit): Commit[] {
+    const commitsBySha: { [sha: string]: Commit[] } =
+      this.applyFiltersToFlattenedCommits()
+    const rows = []
+
+    for (const { branch, sha } of this.rowOrder) {
```

Comment:
[C] Once the table is flattened we could have duplicate entries right next to each other. We will need a way to collapse these rows:

<img width="1728" alt="image" src="https://github.com/iterative/vscode-dvc/assets/37993418/d9b673c4-cb0b-472c-9e42-4791e73b3a1d">

Reply:
The only thing that will differ between records will be the branch/tags. I think we should consider concatenating the information on multiple lines into a single cell.

Reply:
This needs to be fixed before moving on.

Reply:
> This needs to be fixed before moving on.

Oh, ok! Apologies, I misunderstood [your comment](https://github.com/iterative/vscode-dvc/pull/4685#pullrequestreview-1645603063).

Reply:
Fixed in #4735